### PR TITLE
Also add metadata for postDelete hooks triggered from the view

### PR DESCRIPTION
**Pull request by icewind1991**
To make https://github.com/owncloud/core/pull/20865 work properly in all cases

(Yes having two code paths is a PITA and I would really like them gone)

cc @PVince81 @blizzz

### CODE REVIEW

File: lib/private/files/node/hookconnector.php [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 namespace OC\Files\Node;
 
+use Icewind\SMB\IFileInfo;
```
Comment (PVince81) [inline lib/private/files/node/hookconnector.php]: SMB ?!!

Comment (icewind1991) [inline lib/private/files/node/hookconnector.php]: Wrong generated code :speak_no_evil:

Comment (icewind1991) [inline lib/private/files/node/hookconnector.php]: fixed

**Comment by MorrisJobke** [comment]
> (Yes having two code paths is a PITA and I would really like them gone)

Addressing this is only possible by migrating everything to the node API, right?

**Comment by icewind1991** [comment]
> Addressing this is only possible by migrating everything to the node API, right?

The plan it to make the node api work on top of the storage, the view api can be re-implemented on top of the node api for compat

**Comment by blizzz** [comment]
code looks fine :+1:

**Comment by MorrisJobke** [comment]
> The plan it to make the node api work on top of the storage, the view api can be re-implemented on top of the node api for compat

Why do we need this? Isn't it internally only and could be replaced by the node API? Or is it that widely used?

**Comment by icewind1991** [comment]
It's used by a large number of 3rd party apps

**Comment by MorrisJobke** [comment]
> It's used by a large number of 3rd party apps

Can you send around an email to the devel mailing list that all apps should not rely on this and use the public node API. If problems occur with this, then they should report back so that we can properly fix the public node API or enhance it. Can you do this? That would be really nice. Just to bring this into sight of the developers.

**Comment by MorrisJobke** [comment]
Tested and don't throw any error :+1:

**Pull request merged**